### ISSUE TITLE
Why does SFT sum the cross-entropy loss within each sequence?

### ISSUE THREAD

**Issue by yunjae-won** [eric-mitchell/direct-preference-optimization#68]
Thank you for maintaining such an important repository. I really enjoyed and learned a lot from reading your DPO paper.

I have one question regarding the SFT loss implementation in the repository. Apparently, the SFT loss sums the cross entropy loss within each sequences. However, from my understanding, language modeling loss conventionally averages the cross entropy loss for all tokens within the batch [(Ref: GPT2 Loss)](https://github.com/huggingface/transformers/blob/345b9b1a6a308a1fa6559251eb33ead2211240ac/src/transformers/models/gpt2/modeling_gpt2.py#L1106). I think this results in a difference in computing the standard cross entropy loss between TRL's SFTTrainer and this repository's SFT loss. Why is SFT implemented this way?

**Comment by HuXiangkun** [comment]
Same question here. Hi @YJWon99 , do you have any ideas now?

**Comment by cthorrez** [comment]
The losses are averaged 
https://github.com/eric-mitchell/direct-preference-optimization/blob/f8b8c0f49dc92a430bae41585f9d467d3618fe2f/trainers.py#L270

**Comment by yiyepiaoling0715** [comment]
has been solved? same question

**Comment by HuXiangkun** [comment]
@yiyepiaoling0715 I think it's a bug in their code, it should be averaged over the sequence and I made the revision in my experiments.
